src/aura/BGE_ConfigurationWizard/BGE_ConfigurationWizardHelper.js

```diff
@@ -1,1058 +1,545 @@
-                batchInfoView.expectedCount = batchInfo.expectedCount;
-                batchInfoView.expectedTotal = batchInfo.expectedTotal;
-                batchInfoView.recordCount = batchInfo.recordCount;
-
-                // batch processing settings
-                batchInfoView.requireTotalMatch = batchInfo.requireTotalMatch;
-                batchInfoView.batchProcessSize = batchInfo.batchProcessSize;
-                batchInfoView.runOpportunityRollupsWhileProcessing = batchInfo.runOpportunityRollupsWhileProcessing;
-                batchInfoView.donationMatchingBehavior = batchInfo.donationMatchingBehavior;
-                batchInfoView.donationMatchingClass = batchInfo.donationMatchingClass;
-                batchInfoView.donationMatchingOptions = batchInfo.donationMatchingOptions;
-                batchInfoView.donationMatchingRule = batchInfo.donationMatchingRule;
-                batchInfoView.donationDateRange = batchInfo.donationDateRange;
-                batchInfoView.noMatchOnDate = batchInfo.noMatchOnDate;
-                batchInfoView.postProcessClass = batchInfo.postProcessClass;
-
-                component.set('v.batchInfo', batchInfoView);
-            });
-
-            // BatchInfoView module public functions and properties
-            return {
-                name: '',
-                id: '',
-                description: '',
-                expectedCount: 0,
-                expectedTotal: 0,
-                recordCount: 0,
-
-                // batch processing settings
-                requireTotalMatch: false,
-                batchProcessSize: 0,
-                runOpportunityRollupsWhileProcessing: false,
-                donationMatchingBehavior: '',
-                donationMatchingClass: '',
-                donationMatchingOptions: [],
-                donationMatchingRule: [],
-                donationDateRange: '',
-                noMatchOnDate: false,
-                postProcessClass: ''
-            };
-        })(component, model);
+    loadModel: function(component, model) {
+        this.loadBatchInfo(component, model);
+        this.loadWizardMetadata(component, model);
+
+        //parse active fields because it's another level JSON object
+        let activeFields = JSON.parse(model.activeFields);
+        let allFields = model.availableFields;
+        this.loadAvailableFields(component, activeFields, allFields);
     },
 
     /**
-     * @description Returns the Batch Metadata View module.
-     * @param component. Lightning Component reference.
-     * @param model. The Model.
-     * @return View of the Batch Metadata module.
+     * @description: sets basic batch info and processing settings
+     * @param model: parsed JSON model from the model
      */
-    BatchMetadataView : function(component, model) {
-        return (function (component, model) {
+    loadBatchInfo: function(component, model) {
+        let batchInfo = {};
```

Comment:
Optional! I think we could consolidate this as:
```
let batchInfo = model;
batchInfo.expectedCount = batchInfo.expectedCount || 0;
batchInfo.expectedTotal = batchInfo.expectedTotal || 0;
```

Reply:
We would rather not do that here; we left the explicit setting of each attribute because we don't want available fields, active fields, and labels to be written to batchInfo too.

Reply:
If that's the goal, would it make more sense to copy the object and explicitly remove the properties we don't want to include?